### ISSUE TITLE
Playback fails on each launch of the program

### ISSUE THREAD

**Issue by simurq** [DeaDBeeF-Player/deadbeef#2482]
**Deadbeef:** v1.8.4
**OS:** Pop!_OS 20.10

As soon as I double-click any audio from the pre-loaded playlist, the log pops up with Deadbeef trying to locate the file for playback. Below is my log-file.
```starting deadbeef 1.8.4 [static]
server_start
searching for GUI plugins in /home/vna-pc/.local/lib64/deadbeef
searching for GUI plugins in /home/vna-pc/.local/lib/deadbeef
searching for GUI plugins in /opt/deadbeef/lib/deadbeef
load_plugin_dir /opt/deadbeef/lib/deadbeef: scandir found 55 files
found gui plugin ddb_gui_GTK2.so
added GTK2 gui plugin
found gui plugin ddb_gui_GTK3.so
added GTK3 gui plugin
load gui plugin
checking GUI plugin: GTK2
checking GUI plugin: GTK3
found selected GUI plugin: GTK3
loading plugin /opt/deadbeef/lib/deadbeef/ddb_gui_GTK3.so
loading plugins from /home/vna-pc/.local/lib64/deadbeef
loading plugins from /home/vna-pc/.local/lib/deadbeef
loading plugins from /opt/deadbeef/lib/deadbeef
load_plugin_dir /opt/deadbeef/lib/deadbeef: scandir found 55 files
loading plugin /opt/deadbeef/lib/deadbeef/aac.so
loading plugin /opt/deadbeef/lib/deadbeef/adplug.so
loading plugin /opt/deadbeef/lib/deadbeef/alac.so
loading plugin /opt/deadbeef/lib/deadbeef/alsa.so
loading plugin /opt/deadbeef/lib/deadbeef/artwork.so
loading plugin /opt/deadbeef/lib/deadbeef/cdda.so
loading plugin /opt/deadbeef/lib/deadbeef/converter.so
loading plugin /opt/deadbeef/lib/deadbeef/converter_gtk2.so
loading plugin /opt/deadbeef/lib/deadbeef/converter_gtk3.so
loading plugin /opt/deadbeef/lib/deadbeef/dca.so
loading plugin /opt/deadbeef/lib/deadbeef/ddb_dumb.so
loading plugin /opt/deadbeef/lib/deadbeef/ddb_mono2stereo.so
loading plugin /opt/deadbeef/lib/deadbeef/ddb_shn.so
loading plugin /opt/deadbeef/lib/deadbeef/ddb_soundtouch.so
loading plugin /opt/deadbeef/lib/deadbeef/dsp_libsrc.so
loading plugin /opt/deadbeef/lib/deadbeef/ffap.so
loading plugin /opt/deadbeef/lib/deadbeef/ffmpeg.so
loading plugin /opt/deadbeef/lib/deadbeef/flac.so
loading plugin /opt/deadbeef/lib/deadbeef/gme.so
loading plugin /opt/deadbeef/lib/deadbeef/hotkeys.so
loading plugin /opt/deadbeef/lib/deadbeef/in_sc68.so
loading plugin /opt/deadbeef/lib/deadbeef/lastfm.so
loading plugin /opt/deadbeef/lib/deadbeef/m3u.so
loading plugin /opt/deadbeef/lib/deadbeef/mms.so
loading plugin /opt/deadbeef/lib/deadbeef/mp3.so
loading plugin /opt/deadbeef/lib/deadbeef/musepack.so
loading plugin /opt/deadbeef/lib/deadbeef/notify.so
loading plugin /opt/deadbeef/lib/deadbeef/nullout.so
loading plugin /opt/deadbeef/lib/deadbeef/opus.so
loading plugin /opt/deadbeef/lib/deadbeef/oss.so
loading plugin /opt/deadbeef/lib/deadbeef/pltbrowser_gtk2.so
loading plugin /opt/deadbeef/lib/deadbeef/pltbrowser_gtk3.so
loading plugin /opt/deadbeef/lib/deadbeef/psf.so
loading plugin /opt/deadbeef/lib/deadbeef/pulse.so
loading plugin /opt/deadbeef/lib/deadbeef/rg_scanner.so
loading plugin /opt/deadbeef/lib/deadbeef/shellexec.so
loading plugin /opt/deadbeef/lib/deadbeef/shellexecui_gtk2.so
loading plugin /opt/deadbeef/lib/deadbeef/shellexecui_gtk3.so
loading plugin /opt/deadbeef/lib/deadbeef/sid.so
loading plugin /opt/deadbeef/lib/deadbeef/sndfile.so
loading plugin /opt/deadbeef/lib/deadbeef/supereq.so
loading plugin /opt/deadbeef/lib/deadbeef/tta.so
loading plugin /opt/deadbeef/lib/deadbeef/vfs_curl.so
loading plugin /opt/deadbeef/lib/deadbeef/vfs_zip.so
loading plugin /opt/deadbeef/lib/deadbeef/vorbis.so
loading plugin /opt/deadbeef/lib/deadbeef/vtx.so
loading plugin /opt/deadbeef/lib/deadbeef/wavpack.so
loading plugin /opt/deadbeef/lib/deadbeef/wildmidi.so
loading plugin /opt/deadbeef/lib/deadbeef/wma.so
starting plugin GTK3 user interface
starting plugin AAC player
starting plugin Adplug player
starting plugin ALAC player
starting plugin ALSA output plugin
starting plugin Album Artwork
starting plugin Audio CD player
starting plugin Converter
starting plugin Converter GTK2 UI
starting plugin Converter GTK3 UI
starting plugin dts decoder
starting plugin DUMB module player
starting plugin Mono to stereo
starting plugin Shorten player
starting plugin Soundtouch
starting plugin Resampler (Secret Rabbit Code)
starting plugin Monkey's Audio (APE) decoder
starting plugin FLAC decoder
starting plugin Game-Music-Emu player
starting plugin Hotkey manager
starting plugin SC68 player (Atari ST SNDH YM2149)
starting plugin last.fm scrobbler
starting plugin M3U and PLS support
starting plugin mms vfs
starting plugin MP3 player
starting plugin MusePack decoder
starting plugin OSD Notify
starting plugin Null output plugin
starting plugin Opus player
starting plugin OSS output plugin
starting plugin Playlist browser GTK2
starting plugin Playlist browser GTK3
starting plugin PSF player using Audio Overload SDK
starting plugin PulseAudio output plugin
starting plugin ReplayGain Scanner
starting plugin Shell commands
starting plugin Shellexec GTK2 UI
starting plugin Shellexec GTK3 UI
starting plugin SID player
starting plugin WAV/PCM player
starting plugin SuperEQ
starting plugin tta decoder
starting plugin cURL vfs
starting plugin ZIP vfs
starting plugin Ogg Vorbis decoder
starting plugin VTX player
starting plugin WavPack decoder
starting plugin WildMidi player
starting plugin WMA player
starting plugin stdio vfs
starting plugin FFMPEG audio player
selected output plugin: PulseAudio output plugin
plugin Converter GTK2 UI failed to connect to dependencies, deactivated.
plugin Playlist browser GTK2 failed to connect to dependencies, deactivated.
plugin Shellexec GTK2 UI failed to connect to dependencies, deactivated.
Failed to play track: /media/vna-pc/BackupDrive/Musiqi/Детские_песни/Умка.mp3
Failed to play track: /media/vna-pc/BackupDrive/Musiqi/Добрые_песни/Шаланды_полные_кефали.mp3
Failed to play track: /media/vna-pc/BackupDrive/Musiqi/West/Ricchi_E_Poveri_-_Mama_Maria.mp3
Failed to play track: /media/vna-pc/BackupDrive/Musiqi/West/Pink_Floyd_-_Shine_On_Your_Crazy_Diamond.mp3
...
```
The closest reported issue here is #2458 but all my audio files are stored on a separate HDD (BackupDrive), which is linked to Deadbeef through the same playlist loaded on each startup of the program, no other paths. Plus, the `~/.config/deadbeef/config` file has no references to ALSA, as I've configured it to use PulseAudio instead, which works as it should.

Strangely, when I reload the same (!) playlist file, the problem is gone. I don't understand what prevents the program from resolving the issue on its own, each time expecting the user to reload the playlist...

It's also strange to see Deadbeef trying to connect GTK**2** UI to dependencies, as I've assigned GTK**3** UI in the Preferences. These are the only failure messages in the log, AFAIU. 

Thank you and please keep it up! Deadbeef is great indeed!!!

**Comment by Alexey-Yakovenko** [comment]
Are you sure that the external drive is already mounted and readable at the time when you launch deadbeef?
It seems like deadbeef doesn't have access to the files, but then when you re-load playlist access is granted, or perhaps the disk gets mounted.
I suggest to run deadbeef with strace and see if there are any suspicious error messages..

**Comment by Alexey-Yakovenko** [comment]
> It's also strange to see Deadbeef trying to connect GTK2 UI to dependencies, as I've assigned GTK3 UI in the Preferences.

This is normal. There are many plugins which get shipped in both GTK2 and GTK3 version. They need to detect which UI is in use, before activating.

**Comment by simurq** [comment]
> Are you sure that the external drive is already mounted and readable at the time when you launch deadbeef?
> It seems like deadbeef doesn't have access to the files, but then when you re-load playlist access is granted, or perhaps the disk gets mounted.
Gotcha! Your hint was helpful indeed - I moved to the drive through the file manager (to sort of nudge / activate it), then launched Deadbeef without reloading the playlist as I've done earlier, and guess what - playback worked and I didn't have to reload the playlist!
So, this raises another question then - is it the Deadbeef, which doesn't wait until the disk is mounted and ready, or my system, which should make sure the disk is mounted for later usage by applications like Deadbeef?

> I suggest to run deadbeef with strace and see if there are any suspicious error messages..
I did, quite a long file. I'll look for suspicious messages and report back.
